packages/google_maps_flutter/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/GoogleMapsPlugin.java

```diff
@@ -4,196 +4,172 @@
+                    public Lifecycle getLifecycle() {
+                      return ((LifecycleOwner) activity).getLifecycle();
+                    }
+                  }));
+    } else {
+      ProxyLifecycleProvider proxyLifecycleProvider = new ProxyLifecycleProvider(activity);
+      activity.getApplication().registerActivityLifecycleCallbacks(proxyLifecycleProvider);
+      registrar
+          .platformViewRegistry()
+          .registerViewFactory(
+              VIEW_TYPE, new GoogleMapFactory(registrar.messenger(), proxyLifecycleProvider));
+    }
   }
 
   public GoogleMapsPlugin() {}
 
-  private GoogleMapsPlugin(Activity activity) {
-    this.registrarActivityHashCode = activity.hashCode();
-  }
-
   // FlutterPlugin
 
   @Override
   public void onAttachedToEngine(FlutterPluginBinding binding) {
-    pluginBinding = binding;
+    binding
+        .getPlatformViewRegistry()
+        .registerViewFactory(VIEW_TYPE, new GoogleMapFactory(binding.getBinaryMessenger(), this));
   }
 
   @Override
-  public void onDetachedFromEngine(FlutterPluginBinding binding) {
-    pluginBinding = null;
-  }
+  public void onDetachedFromEngine(FlutterPluginBinding binding) {}
 
   // ActivityAware
 
   @Override
   public void onAttachedToActivity(ActivityPluginBinding binding) {
     lifecycle = FlutterLifecycleAdapter.getActivityLifecycle(binding);
-    lifecycle.addObserver(this);
-    pluginBinding
-        .getPlatformViewRegistry()
-        .registerViewFactory(
-            VIEW_TYPE,
-            new GoogleMapFactory(
-                state,
-                pluginBinding.getBinaryMessenger(),
-                binding.getActivity().getApplication(),
-                lifecycle,
-                null));
   }
 
   @Override
   public void onDetachedFromActivity() {
-    lifecycle.removeObserver(this);
     lifecycle = null;
   }
 
-  @Override
-  public void onDetachedFromActivityForConfigChanges() {
-    this.onDetachedFromActivity();
-  }
-
   @Override
   public void onReattachedToActivityForConfigChanges(ActivityPluginBinding binding) {
-    lifecycle = FlutterLifecycleAdapter.getActivityLifecycle(binding);
-    lifecycle.addObserver(this);
-  }
-
-  // DefaultLifecycleObserver methods
-
-  @Override
-  public void onCreate(@NonNull LifecycleOwner owner) {
-    state.set(CREATED);
+    onAttachedToActivity(binding);
   }
 
   @Override
-  public void onStart(@NonNull LifecycleOwner owner) {
-    state.set(STARTED);
-  }
-
-  @Override
-  public void onResume(@NonNull LifecycleOwner owner) {
-    state.set(RESUMED);
+  public void onDetachedFromActivityForConfigChanges() {
+    onDetachedFromActivity();
   }
 
+  @Nullable
   @Override
-  public void onPause(@NonNull LifecycleOwner owner) {
-    state.set(STARTED);
+  public Lifecycle getLifecycle() {
+    return lifecycle;
   }
 
-  @Override
-  public void onStop(@NonNull LifecycleOwner owner) {
-    state.set(CREATED);
-  }
+  private static final class ProxyLifecycleProvider
```

Comment:
nit: Can you add some comments here describing that this is used make a lifecycle our of the application lifecycle calls?

Reply:
Done